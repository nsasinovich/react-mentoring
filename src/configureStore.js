import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';
import { fromJS } from 'immutable';

import appReducer from './reducers/app_reducer';
import { rootSaga } from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

export default (initialState) => {
    const store = createStore(appReducer, fromJS(initialState), applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};

