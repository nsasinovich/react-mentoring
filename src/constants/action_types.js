export default {
    CHANGE_SORT: 'CHANGE_SORT',
    UPDATE_SEARCH_INPUT: 'UPDATE_SEARCH_INPUT',
};
