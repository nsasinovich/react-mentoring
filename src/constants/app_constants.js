export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};

export const FilterOptions = {
    GENRE: 'genres',
    TITLE: 'title',
};

export const SortingOptions = {
    RELEASE: {
        name: 'release date',
        sortField: 'year',
    },
    RATING: {
        name: 'rating',
        sortField: 'vote_count',
    },
};

export const INITIAL_STATE = {
    results: [],
    resultsCount: 0,
    searchInput: '',
    selectedSort: SortingOptions.RATING,
    selectedFilter: FilterOptions.TITLE,
    selectedMovieDetails: null,
};
