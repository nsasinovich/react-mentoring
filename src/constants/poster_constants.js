export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};
