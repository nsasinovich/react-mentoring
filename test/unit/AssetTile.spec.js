import { shallow } from 'enzyme';
import React from 'react';
import configureStore from 'redux-mock-store';
import { PosterSizes, INITIAL_STATE } from 'src/constants/app_constants';
import AssetTile from 'src/components/asset_tile/asset_tile';
import Poster from 'src/components/shared/poster/poster';

const mockStore = configureStore();

const assetMock = {
    poster_path: 'poster_path',
    genres: ['genre1', 'genre2'],
    release_date: '2017-06-06',
};

let store;
let wrapper;

describe('<AssetTile />', () => {
    beforeEach(() => {
        store = mockStore(INITIAL_STATE);
        wrapper = shallow(<AssetTile asset={assetMock} store={store}/>).dive();
    });

    it('should render AssetTile and match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain Poster element', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<Poster posterSize={PosterSizes.MEDIUM} imgUrl={assetMock.poster_path} />)).toBe(true);
    });
});
