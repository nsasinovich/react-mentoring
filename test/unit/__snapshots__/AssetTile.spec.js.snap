// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetTile /> should contain Poster element 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AssetTile
    asset={
      Object {
        "genres": Array [
          "genre1",
          "genre2",
        ],
        "poster_path": "poster_path",
        "release_date": "2017-06-06",
      }
    }
    showMovieDetails={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Poster
          imgUrl="poster_path"
          posterSize="poster-medium"
        />,
        <div
          className="asset-info"
        >
          <div>
            <p
              className="title"
              onClick={[Function]}
            />
            <p
              className="genre"
            >
              genre1
            </p>
          </div>
          <p
            className="year"
          >
            2017
          </p>
        </div>,
      ],
      "className": "asset-tile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "imgUrl": "poster_path",
          "posterSize": "poster-medium",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <p
                className="title"
                onClick={[Function]}
              />
              <p
                className="genre"
              >
                genre1
              </p>
            </div>,
            <p
              className="year"
            >
              2017
            </p>,
          ],
          "className": "asset-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="title"
                  onClick={[Function]}
                />,
                <p
                  className="genre"
                >
                  genre1
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "title",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "genre1",
                  "className": "genre",
                },
                "ref": null,
                "rendered": "genre1",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2017,
              "className": "year",
            },
            "ref": null,
            "rendered": 2017,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Poster
            imgUrl="poster_path"
            posterSize="poster-medium"
          />,
          <div
            className="asset-info"
          >
            <div>
              <p
                className="title"
                onClick={[Function]}
              />
              <p
                className="genre"
              >
                genre1
              </p>
            </div>
            <p
              className="year"
            >
              2017
            </p>
          </div>,
        ],
        "className": "asset-tile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "imgUrl": "poster_path",
            "posterSize": "poster-medium",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <p
                  className="title"
                  onClick={[Function]}
                />
                <p
                  className="genre"
                >
                  genre1
                </p>
              </div>,
              <p
                className="year"
              >
                2017
              </p>,
            ],
            "className": "asset-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title"
                    onClick={[Function]}
                  />,
                  <p
                    className="genre"
                  >
                    genre1
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "genre1",
                    "className": "genre",
                  },
                  "ref": null,
                  "rendered": "genre1",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2017,
                "className": "year",
              },
              "ref": null,
              "rendered": 2017,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<AssetTile /> should render AssetTile and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AssetTile
    asset={
      Object {
        "genres": Array [
          "genre1",
          "genre2",
        ],
        "poster_path": "poster_path",
        "release_date": "2017-06-06",
      }
    }
    showMovieDetails={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Poster
          imgUrl="poster_path"
          posterSize="poster-medium"
        />,
        <div
          className="asset-info"
        >
          <div>
            <p
              className="title"
              onClick={[Function]}
            />
            <p
              className="genre"
            >
              genre1
            </p>
          </div>
          <p
            className="year"
          >
            2017
          </p>
        </div>,
      ],
      "className": "asset-tile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "imgUrl": "poster_path",
          "posterSize": "poster-medium",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <p
                className="title"
                onClick={[Function]}
              />
              <p
                className="genre"
              >
                genre1
              </p>
            </div>,
            <p
              className="year"
            >
              2017
            </p>,
          ],
          "className": "asset-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="title"
                  onClick={[Function]}
                />,
                <p
                  className="genre"
                >
                  genre1
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "title",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "genre1",
                  "className": "genre",
                },
                "ref": null,
                "rendered": "genre1",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2017,
              "className": "year",
            },
            "ref": null,
            "rendered": 2017,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Poster
            imgUrl="poster_path"
            posterSize="poster-medium"
          />,
          <div
            className="asset-info"
          >
            <div>
              <p
                className="title"
                onClick={[Function]}
              />
              <p
                className="genre"
              >
                genre1
              </p>
            </div>
            <p
              className="year"
            >
              2017
            </p>
          </div>,
        ],
        "className": "asset-tile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "imgUrl": "poster_path",
            "posterSize": "poster-medium",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <p
                  className="title"
                  onClick={[Function]}
                />
                <p
                  className="genre"
                >
                  genre1
                </p>
              </div>,
              <p
                className="year"
              >
                2017
              </p>,
            ],
            "className": "asset-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title"
                    onClick={[Function]}
                  />,
                  <p
                    className="genre"
                  >
                    genre1
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "genre1",
                    "className": "genre",
                  },
                  "ref": null,
                  "rendered": "genre1",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2017,
                "className": "year",
              },
              "ref": null,
              "rendered": 2017,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
